src/cart/swim/swim_internal.h

```diff
@@ -65,11 +65,9 @@ extern "C" {
 #endif
 
 /** SWIM protocol parameter defaults */
-#define SWIM_PROTOCOL_PERIOD_LEN 2400	/* milliseconds, should NOT be less
-					 * than 3 * SWIM_PING_TIMEOUT
-					 */
-#define SWIM_SUSPECT_TIMEOUT	3 * SWIM_PROTOCOL_PERIOD_LEN
-#define SWIM_PING_TIMEOUT	800	/* milliseconds */
+#define SWIM_PROTOCOL_PERIOD_LEN 1000	/* milliseconds */
+#define SWIM_SUSPECT_TIMEOUT	(8 * SWIM_PROTOCOL_PERIOD_LEN)
+#define SWIM_PING_TIMEOUT	900	/* milliseconds */
```

Comment:
Sorry, not following why the period can be so close to the ping timeout now... If a dping sent from rank x to rank y times out, rank x may not have learned any sms_delay info about rank y from others yet. So the following iping may still need > 2 ping timeouts to time out. But there's only 1000 - 900 ms left until this period ends. That is, ..., not the case?

Reply:
My SWIM implementation is state machine. So, it stay on Dping or Iping state until ACK or timeout. Then it check if time to start next period. If ACK was received very quick it will wait otherwise it starts new period. So, period time here is just lower bound for period duration (new message cannot be sent early than this period time passed).

Reply:
So in the iping scenario I described above, even when the 1000 ms period passes, rank x will still wait for the iping ACKs/timeouts, before starting a new period?

Reply:
In the begin of period SWIM algorithm D-pings target and then wait for SWIM_PING_TIMEOUT. If no ACK received during this time it sends two I-ping requests to other members and then wait for 2 * SWIM_PING_TIMEOUT. If no guesses about suspected member received during this time it select new target and start new period. Therefore the new period will be started in between of SWIM_PROTOCOL_PERIOD_LEN and 3 * SWIM_PING_TIMEOUT.

Reply:
I see. Thanks.